torch_geometric/data/data.py

```diff
@@ -842,6 +842,9 @@ def _put_edge_index(self, edge_index: EdgeTensorType,
         attr_val = edge_tensor_type_to_adj_type(edge_attr, edge_index)
         setattr(self, attr_name, attr_val)
 
+        # Set edge attributes:
+        setattr(self, f'{attr_name}_edge_attr', edge_attr)
```

Comment:
Also set `size` of `edge_attr` (if not specified). WDYT?

Reply:
Not sure I understand: why would it be useful to set the size of `EdgeAttr` here? We set the size of the `Data` object if it's provided in `edge_attr`, and when we `get_all_edge_attrs` we return this size as well.

Reply:
Then I think we need some clarity on what we actually want to save in `EdgeAttr` - I was under the impression that all its information is useful to maintain. If the `EdgeAttr` does not define `size`, we can set it based on `self.num_nodes`.